### ISSUE TITLE
top-scope variable check incorrectly warning on second parameter in block

### ISSUE THREAD

**Issue by jstangroome** [rodjek/puppet-lint#380]
```
WARNING: top-scope variable being used without an explicit namespace on line K

  $array.each |$index, $element| { something($element) }
                                             ^
```

Doesn't trigger on first parameter, eg `$index`. Also happens with `reduce` function.

**Comment by rodjek** [comment]
@jstangroome what version of puppet-lint are you running? I'm not able to reproduce this with that snippet locally with 1.1.0.

**Comment by gburjan** [comment]
We are facing with a similar issue. With the following code I can produce the same warning using puppet-lint 1.1.0.

    [gburjan@orion manifests]$ gem list puppet-lint
 
    *** LOCAL GEMS ***
 
    puppet-lint (1.1.0)
    [gburjan@orion manifests]$ cat test.pp
    # This is test::test class to reproduce puppet-lint issue #380
 
    class test::test {
      $hash = { foo => { bar => 'baz' } }
      $hash.each |$key, $value| {
        $value['bar']
      }
    }
    [gburjan@orion manifests]$ puppet lint test.pp
    WARNING: top-scope variable being used without an explicit namespace on line 6
    [gburjan@orion manifests]$

**Comment by href** [comment]
I see the same problem. But only if I access the hash like in the example above: 
`$value['bar']` triggers the warning, but `$value` doesn't.

**Comment by mmessmore** [comment]
Just noting that I keep running into this as well in the same situation that href is speaking of.  

WARNING: top-scope variable being used without an explicit namespace on line 12

    class bar() {

      $baz_hash = {
        test  => {
          foo => nunchucks,
        },
      }

      $baz_hash.each |$key, $value| {
        notify {$key:}
        notify {$value:}
        notify {$value['foo']:}
      }
    }

**Comment by hugoboos** [comment]
Hi,

Same problem here. Using version 1.1.0.

```puppet
class test {
  $environment_variables = {
    test => {
      name => 'test',
      value => '123'
    }
  }

  $test = $environment_variables.map |String $key, Hash $variable| {
    "set ${variable[name]} '${variable[value]}'"
  }

  notice($test)
}

include ::test
```

```
WARNING: top-scope variable being used without an explicit namespace on line 10
WARNING: top-scope variable being used without an explicit namespace on line 10
```

**Comment by rvincentatprokarmadotcom** [comment]
I can confirm:

```
rvincent@atllvcidap030p:~ puppet-lint --version
puppet-lint 1.1.0
rvincent@atllvcidap030p:~ cat > test.pp
class test {
  $environment_variables = {
    test => {
      name => 'test',
      value => '123'
    }
  }

  $test = $environment_variables.map |String $key, Hash $variable| {
    "set ${variable[name]} '${variable[value]}'"
  }

  notice($test)
}

include ::test
rvincent@atllvcidap030p:~ puppet-lint test.pp
ERROR: test not in autoload module layout on line 1
WARNING: top-scope variable being used without an explicit namespace on line 10
WARNING: top-scope variable being used without an explicit namespace on line 10
WARNING: class not documented on line 1
```

**Comment by rvincentatprokarmadotcom** [comment]
Duplicate issue rodjek#450 includes a fix.  See pull request rodjek#453

**Comment by rnelson0** [comment]
Close as duplicate of #450
